packages/typespec-ts/test/unit/modelsGenerator.spec.ts

```diff
@@ -847,6 +840,448 @@ describe("Input/output model type", () => {
+      assert.strictEqual(outputModelFile?.path, "outputModels.ts");
+      assertEqualContent(
+        outputModelFile?.content!,
+        `
+        export interface VegetablesOutput {
+          carrots: VegetableCarrotOutput;
+          beans: VegetableBeansOutput;
+        }
+        
+        export interface VegetableCarrotOutput extends Record<string, string> {}
+        
+        export interface VegetableBeansOutput extends Record<string, string> {}
+        `
+      );
+    });
+
+    it("should handle model additional properties extends from record of object", async () => {
+      const schemaOutput = await emitModelsFromTypeSpec(`
+      model VegetableCarrot extends Record<Carrots> {
+        testProp: Carrots
+      }
+      model VegetableBeans extends Record<Beans> {}
+      
+      model Vegetables {
+        carrots: VegetableCarrot,
+        beans: VegetableBeans
+      }
+      model Carrots {
+        color: string,
+        id: string
+      }
+      model Beans {
+        expiry: string,
+        id: string
+      }
+      op post(@body body: Vegetables): { @body body: Vegetables };
+      `);
+      assert.ok(schemaOutput);
+      const { inputModelFile, outputModelFile } = schemaOutput!;
+      assert.ok(inputModelFile);
+      assert.strictEqual(inputModelFile?.path, "models.ts");
+      assertEqualContent(
+        inputModelFile?.content!,
+        `
+        export interface Vegetables {
+          carrots: VegetableCarrot;
+          beans: VegetableBeans;
+        }
+        
+        export interface VegetableCarrot extends Record<string, Carrots> {
+          testProp: Carrots;
+        }
+
+        export interface Carrots {
+          color: string;
+          id: string;
+        }
+        
+        export interface VegetableBeans extends Record<string, Beans> {}
+        
+        export interface Beans {
+          expiry: string;
+          id: string;
+        }`
+      );
+
+      assert.ok(outputModelFile);
+      assert.strictEqual(outputModelFile?.path, "outputModels.ts");
+      assertEqualContent(
+        outputModelFile?.content!,
+        `
+        export interface VegetablesOutput {
+          carrots: VegetableCarrotOutput;
+          beans: VegetableBeansOutput;
+        }
+        
+        export interface VegetableCarrotOutput extends Record<string, CarrotsOutput> {
+          testProp: CarrotsOutput;
+        }
+
+        export interface CarrotsOutput {
+          color: string;
+          id: string;
+        }
+        
+        export interface VegetableBeansOutput extends Record<string, BeansOutput> {}
+        
+        export interface BeansOutput {
+          expiry: string;
+          id: string;
+        }`
+      );
+    });
+
+    it("should handle model additional properties is from record of object", async () => {
+      const schemaOutput = await emitModelsFromTypeSpec(`
+      model VegetableCarrot is Record<Carrots> {
+        testProp: Carrots
+      }
```

Comment:
not sure if it's appropriate to generate Vegetables as 
```ts
        export interface Vegetables {
          carrots: Record<string, Carrots>;
          beans: Record<string, Beans>;
        }
```

Reply:
@joheredi in the case of is keyword being used in record to express additional properties,
we currently just drop the model `VegetableCarrot` generation, use the `Record<string, Carrots>` to replace it. which has information loss about VegetableCarrot's own properties. And according to Timothee's comment here https://github.com/Azure/cadl-ranch/pull/330#issuecomment-1555035443 we should not treat extends and is as the same thing, 
One possible solution here is to use indexer syntax here to express additional properties for is Record scenario. like we should generate model VegetableCarrot as 
```ts
export interface VegetableCarrot {
  [prop: string]: Carrots;
  testProp: Carrots;
}
```
How do you think ? feel free to come up with different solutions here :)

Reply:
I think we should still use Record when setting additional properties

```ts
export interface VegetableCarrot {
  [prop: string]: Carrots;
  testProp: Carrots;
}
```

is identical to

```ts
export interface VegetableCarrot extends Record<string, Carrots> {
  testProp: Carrots;
}
```

But using records is better for documenting. If I understand correctly Timothee's comment is in the context of TypeSpec's `is` vs `extends` which I think makes sense in that context and will have implications when emitting OpenAPI, but I also think modeling the TypeScript interface extending record is the best way we can represent it for our customers.



When additional properties are not in the picture then if we saw

```ts
model Foo is Record<Carrots>;
```

we  should generate

```ts
type Foo = Record<string, Carrots>;
```

and for 

```ts
model Foo extends Carrots {
   foo: string;
}
```

```ts
interface Foo extends Carrots {
   foo: string
}
```

Reply:
it's a little strange to handle `is` differently based on whether there's extra properties in that model.
Another concern is, this is how we handle `is` in normal model context.
```
model Foo is Carrots {
  prop: string;
};
model Carrots {
  prop1: numeric;
};
```
and we will not generate model Carrots, just generate model Foo like 
```ts
interface Foo {
   prop: string;
   prop1: number;
}
```